crmd/join_client.c

```diff
@@ -177,6 +177,32 @@ join_query_callback(xmlNode * msg, int call_id, int rc, xmlNode * output, void *
     free_xml(generation);
 }
 
+static void
+set_join_state(const char * start_state)
+{
+    if (safe_str_eq(start_state, "standby")) {
+        char *attr_id = crm_strdup_printf("nodes-%.256s-standby", fsa_our_uuid);
+        crm_notice("Forcing node %s to join in %s state per configured environment",
+                   fsa_our_uname, start_state);
+        update_attr_delegate(fsa_cib_conn, cib_sync_call, XML_CIB_TAG_NODES, fsa_our_uuid,
+                             NULL, NULL, attr_id, "standby", "on", TRUE, NULL, NULL);
```

Comment:
We don't actually need to create attr_id. We can just pass NULL to update_attr_delegate(), and it will check whether there is an existing attribute, and use its ID if so, otherwise it will create one.